src/server/coprocessor/endpoint.rs

```diff
@@ -215,20 +215,23 @@ impl BatchRunnable<Task> for Host {
                             continue;
                         }
                     };
-                    let end_point = TiDbEndPoint::new(snap);
+                    let len = reqs.len() as f64;
```

Comment:
Good idea, this PR looks good to me generally.  A better choice is to separate the query by point query and complex query, we can improve the case later.